### ISSUE TITLE
noMisusedPromises: incorrect 'await' suggestion when union type includes falsy values (e.g. Promise<T> | null)

### ISSUE THREAD

**Issue by ceifa** [biomejs/biome#9372]
## Description

PR #8495 introduced a regression in the `noMisusedPromises` rule (specifically the `checkConditionals` option).

When a value has a type of `Promise<T> | null` (or similar unions with solid falsy values like `undefined`), using it in a simple `if (value)` check is a valid and idiomatic way to check for nullability. The rule now incorrectly reports this and suggests adding `await`, which changes the semantics of the code.

## Steps to reproduce

```ts
async function example(): Promise<string> | null {
  return null;
}

const value = example();

if (value) { // <- noMisusedPromises fires here, suggesting await
  // ...
}
```

## Expected behavior

The rule should **not** suggest `await` when the union type includes a solid falsy value (e.g. `null`, `undefined`, `false`). In those cases, the `if (value)` check is a valid null/falsy guard, not a misused Promise.

A fix could check whether the non-Promise members of the union are all falsy types before reporting.

## Actual behavior

The rule fires and suggests `await`, which is incorrect. The user can work around it with `if (value !== null)`, but the `await` suggestion is misleading.

## Environment

- Introduced by: #8495
- Rule: `noMisusedPromises` (`checkConditionals`)
